meshnets/modules/lightning_wrapper.py

```diff
@@ -8,12 +8,39 @@
 class MGNLightningWrapper(pl.LightningModule):
     """Simple pytorch_lightning wrapper for the MeshGraphNet training logic."""
 
-    def __init__(self, model: torch.nn.Module, learning_rate: float = 1e-3):
+    def __init__(self,
+                 model: torch.nn.Module,
+                 data_stats: dict,
+                 learning_rate: float = 1e-3):
         super().__init__()
         self.model = model
+        self.data_stats = data_stats
         self.learning_rate = learning_rate
 
+    def normalize(self, batch: Batch) -> Batch:
+
+        # Normalize the input features
+        batch.x = (batch.x -
+                   self.data_stats['x_mean']) / self.data_stats['x_std']
+
+        batch.edge_attr = (batch.edge_attr - self.data_stats['edge_attr_mean']
+                          ) / self.data_stats['edge_attr_std']
```

Comment:
Isn't this standard `Batch Normalization`. Could we not replace this by a batch normalization layer. This has an implementation for [torch geometric](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/norm/batch_norm.html).

Also, I think it should be:

```
batch.x = (batch.x - self.data_stats['x_mean']) / (self.data_stats['x_std'] + epsilon)
```

where `epsilon` is a small constant for added numerical stability

Reply:
Additionally batch normalization is difference during training and inference. From the [original paper](https://arxiv.org/pdf/1502.03167.pdf).

*The normalization of activations that
depends on the mini-batch allows efficient training, but is
neither necessary nor desirable during inference; we want
the output to depend only on the input, deterministically.
For this, once the network has been trained, we use the
normalization*

And here I am not seeing difference in the inference step which can cause high performance models to look poorly trained.

The default batch normalization layers do this automatically when we call `model.eval()`

Reply:
This is not really like batch normalization. What we do here is normalise all the data with respect to the training set statistics. This way, the training set features and labels will have 0 mean and 1 variance, it does not apply at batch level. 

We then have to normalise the test set in the same way otherwise the features between train and test will have completely different ranges.